### PR TITLE
Feature/vtk base module

### PR DESCRIPTION
**Pull request by r-englund**
* Moved VTK module from research-dev
* Cherry-picked commit from #29 
* Fixes compile issues with Property assigment operator (inviwo/inviwo#646)
* Moved the VTK logger to the VTK module 

**Note:** Tested on my machine at home with VTK 6.3.0
- [x]  Will test on my office machine where I have latest VTK (9-something) tomorrow morning

### CODE REVIEW

File: misc/vtk/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# Create module
+ivw_create_module(${SOURCE_FILES} ${HEADER_FILES})
+
+if (VTK_VERSION VERSION_LESS "8.90.0")
+    include(${VTK_USE_FILE})
+    target_link_libraries(${${mod}_target} PUBLIC ${VTK_LIBRARIES})
+    target_include_directories(${${mod}_target} PUBLIC ${VTK_INCLUDE_DIRS})
+else ()
+    target_link_libraries(${${mod}_target} PUBLIC ${VTK_LIBRARIES})
+    vtk_module_autoinit(TARGETS ${${mod}_target} MODULES ${VTK_LIBRARIES})
```
Comment (r-englund) [inline misc/vtk/CMakeLists.txt]: might need to add target_include_directories in the else statement as well, will test that on my office computer tomorrow

**Comment by r-englund** [comment]
Ready for review

File: misc/vtk/include/inviwo/vtk/util/lambda2func.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+namespace inviwo {
+template <typename Callable>
+union storage {
+    storage() {}
+    ~storage() {}
+    std::decay_t<Callable> callable;
+};
+
+template <int, typename Callable, typename Ret, typename... Args>
+auto fnptr_(Callable&& c, Ret (*)(Args...)) {
```
Comment (petersteneteg) [inline misc/vtk/include/inviwo/vtk/util/lambda2func.h]: what is this? why don't we just used std::function

Comment (r-englund) [reply misc/vtk/include/inviwo/vtk/util/lambda2func.h]: Used in https://github.com/inviwo/modules/blob/master/tensorvis/tensorvisio/src/processors/vtkreader.cpp and https://github.com/inviwo/modules/blob/master/tensorvis/tensorvisio/src/processors/vtkunstructuredgridtorectilineargrid.cpp 
I just tried to change from fnptr to std::function but then it does not compile

Comment (r-englund) [reply misc/vtk/include/inviwo/vtk/util/lambda2func.h]:
```
Error	C2664	 'void vtkCallbackCommand::SetCallback(void (__cdecl *)(vtkObject *,unsigned long,void *,void *))': cannot convert argument 1 from 'std::function<void (vtkObject *,unsigned long,void *,void *)>' to 'void (__cdecl *)(vtkObject *,unsigned long,void *,void *)'	inviwo-module-tensorvisio	E:\github-master-repo\inviwo\additional\modules\tensorvis\tensorvisio\src\processors\vtkreader.cpp	161	
 ```

**Pull request merged**